Classes/DashboardConfiguration.php

```diff
@@ -0,0 +1,172 @@
+        }
+        $this->firstLevelCacheDashboards = $dashboards;
+        return $dashboards;
+    }
+
+    /**
+     * Resolve all widgets objects which have been found in the filesystem.
+     *
+     * @param bool $useCache
+     * @return Widget[]
+     */
+    protected function resolveAllExistingWidgets(bool $useCache = true): array
+    {
+        $widgets = [];
+        $dashboardConfiguration = $this->getAllDashboardConfigurationFromFiles($useCache);
+        foreach ($dashboardConfiguration['Dashboard']['Widgets'] ?? [] as $configuration) {
+            $widgets[$configuration['identifier']] = GeneralUtility::makeInstance(Widget::class, $configuration);
+        }
+        $this->firstLevelCacheWidgets = $widgets;
+        return $widgets;
+    }
+
+    /**
+     * Read the dashboard configuration from config files.
+     *
+     * @param bool $useCache
+     * @return array
+     * @throws NoSuchCacheException
+     */
+    protected function getAllDashboardConfigurationFromFiles(bool $useCache = true): array
+    {
+        // Check if the data is already cached
+        if ($useCache && $dashboardConfiguration = $this->getCache()->get($this->cacheIdentifier)) {
+            // Due to the nature of PhpFrontend, the `<?php` and `#` wraps have to be removed
+            $dashboardConfiguration = preg_replace('/^<\?php\s*|\s*#$/', '', $dashboardConfiguration);
+            $dashboardConfiguration = json_decode($dashboardConfiguration, true);
+        }
+
+        if (empty($dashboardConfiguration)) {
+            $paths = [];
+            foreach ($this->packageManager->getActivePackages() as $package) {
+                $tmpPath = $package->getPackagePath() . 'Configuration/';
+                if (file_exists($tmpPath)) {
+                    $paths[] = $tmpPath;
+                }
+            }
+
+            $finder = new Finder();
+            try {
+                $finder->files()->depth(0)->in($paths)->name($this->configFileName);
+            } catch (\InvalidArgumentException $e) {
+                $finder = [];
+            }
+            $loader = GeneralUtility::makeInstance(YamlFileLoader::class);
+            $dashboardConfiguration = [];
+            foreach ($finder as $fileInfo) {
+                $configuration = $loader->load(GeneralUtility::fixWindowsFilePath((string)$fileInfo));
+                ArrayUtility::mergeRecursiveWithOverrule($dashboardConfiguration, $configuration);
+            }
+            $this->getCache()->set($this->cacheIdentifier, json_encode($dashboardConfiguration));
+        }
+        return $dashboardConfiguration ?? [];
+    }
+
+    /**
+     * Short-hand function for the cache
+     *
+     * @return FrontendInterface
+     * @throws NoSuchCacheException
+     */
+    protected function getCache(): FrontendInterface
+    {
+        return GeneralUtility::makeInstance(CacheManager::class)->getCache('core');
```

Comment:
This is v10 only. In v9 it is cache_core. Would be great if we can check both in this case.

Reply:
good catch, I will push an update